Gems/Atom/Feature/Common/Assets/Shaders/Materials/BasePBR/BasePBR_LightingData.azsli

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// This #define magic lets you use the LightingData class in this file without making it the final LightingData
+// used in your shader. Simply #define LightingData to your custom definition before including this file
+// 
+#ifndef LightingData
+#define LightingData        LightingData_BasePBR
```

Comment:
It may be useful to keep a convention that macros should be always capitalized `LIKE_SO` so that it's clear what is and isn't a preprocessor definition. I get that in the implementation these are used like structs/functions here and other places, but some sort of convention that advertises "hey this symbol is replaceable" may make sense.

I am wondering, however, if there's a way to use `interface` or `class` to achieve this sort of dispatch without needing to rely on macro substitution

Reply:
Could consider for a follow up. For this review changing LightingData or Surface would touch too many files (this PR is already too long), which is why I didn't rename Surface to SurfaceData as Chris suggested above. A standalone renaming PR would be easy to test and easy to review 🙂